osbs/build/build_request.py

```diff
@@ -465,21 +467,63 @@ def set_kerberos_auth(self, plugins):
             self.dj.dock_json_set_arg(phase, plugin, 'koji_principal', krb_principal)
             self.dj.dock_json_set_arg(phase, plugin, 'koji_keytab', krb_keytab)
 
-    def set_reactor_config_map(self):
-        if not self.spec.reactor_config_map.value:
+    def set_reactor_config(self):
+        if not self.spec.reactor_config_map.value and not self.spec.reactor_config_override.value:
             return
         custom = self.template['spec']['strategy']['customStrategy']
 
-        reactor_config_map = {
-            'name': 'REACTOR_CONFIG',
-            'valueFrom': {
-                'configMapKeyRef': {
-                    'name': self.spec.reactor_config_map.value,
-                    'key': 'config.yaml'
+        if self.spec.reactor_config_override.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'value': yaml.dump(self.spec.reactor_config_override.value)
+            }
+        elif self.spec.reactor_config_map.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'valueFrom': {
+                    'configMapKeyRef': {
+                        'name': self.spec.reactor_config_map.value,
+                        'key': 'config.yaml'
+                    }
                 }
             }
-        }
-        custom['env'].append(reactor_config_map)
+
+        custom['env'].append(reactor_config)
+
+    def set_required_secrets(self):
+        """
+        Sets required secrets for worker build
+        """
+        if self.spec.platforms.value is not None:
+            return
+        if not self.spec.reactor_config_override.value:
+            return
+
+        data =self.spec.reactor_config_override.value
+        required_secrets = data.get('required_secrets')
+        if not required_secrets :
+            return
+
+        if 'secrets' not in self.template['spec']['strategy']['customStrategy']:
+            self.template['spec']['strategy']['customStrategy']['secrets'] = []
+
+        custom = self.template['spec']['strategy']['customStrategy']
+        for secret in required_secrets:
+            secret_path = os.path.join(SECRETS_PATH, secret)
+
+            logger.info("Configuring %s secret at %s", secret, secret_path)
+            existing = [secret_mount for secret_mount in custom['secrets']
+                        if secret_mount['secretSource']['name'] == secret]
+
+            if existing:
+                logger.debug("secret %s already set", secret)
+            else:
```

Comment:
Could use `continue` on the if-block above to avoid nested code.

Reply:
ok